safesocket/safesocket_darwin.go

```diff
@@ -83,6 +86,11 @@ func localTCPPortAndTokenDarwin() (port int, token string, err error) {
 				}
 			}
 		}
+		if os.Getuid() == 0 {
+			warnAboutRootOnce.Do(func() {
```

Comment:
Add a comment here too about why this exists, the rest of the file is very verbose on comments might as well continue that here.